pkg/services/alerting/engine.go

```diff
@@ -202,6 +204,18 @@ func (e *AlertingService) processJob(attemptID int, attemptChan chan int, cancel
 		}
 
 		evalContext.Rule.State = evalContext.GetNewState()
+		if evalContext.Rule.State == m.AlertStateOK && evalContext.PrevAlertState != m.AlertStateOK {
```

Comment:
I think this code fits better in the result_handler before it tries to send all alert notifications https://github.com/grafana/grafana/blob/master/pkg/services/alerting/result_handler.go#L90